plugins/dialog/plugin.js

```diff
@@ -300,6 +300,9 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 			definition: definition
 		}, editor ).definition;
 
+		// (#2277)
+		setConfigDefaultValues( this );
```

Comment:
This is a subjective thing, but I'd rather like to see this method be a private one, defined in prototype of CKEditor.dialog type. Then the function would contain helper functions just like you did here. The difference would be there would be much closer to the place where used, so that would make the code more concise. Now we end up with tens of helper functions one after the other, which might not be related to each other.

Reply:
Make sure to do proper api docs for this method.